### ISSUE TITLE
Resizing on Linux causes window to move

### ISSUE THREAD

**Issue by timsneath** [flutter/flutter#53457]
## Steps to Reproduce

 - On Ubuntu, run `flutter create bug` and then `flutter run`.
 - Mouse over the resize handle on the bottom left or top left corner
 - Resize the window until it has zero width, and then continue to resize to the right
 - The window will 'jump' further right

## Expected

 - Window should probably not be resizable to 0 pixels wide
 - Alternative, window should not be resizable to a negative pixel dimension

## Flutter build
```
timsneath@ubuntu:~/scratch/linux$ flutter doctor -v
[✓] Flutter (Channel master, v1.16.4-pre.13, on Linux, locale en_US.UTF-8)
    • Flutter version 1.16.4-pre.13 at
      /home/timsneath/snap/flutter/common/flutter
    • Framework revision 94500073a0 (2 hours ago), 2020-03-27 16:36:15 -0700
    • Engine revision 91ccdbe7f1
    • Dart version 2.8.0 (build 2.8.0-dev.18.0 1402e8e1a4)

[✓] Linux toolchain - develop for Linux desktop
    • clang++ 6.0.0
    • GNU Make 4.1
```

**Comment by stuartmorgan** [comment]
Is this a behavior of your window manager? We're not seeing it on gLinux.

We could set a default minimum size I guess, but to me that feels like something either the window manager should do (e.g., macOS does this, and apparently at least some Linux window managers) based on their default window controls, or the app developer should do based on their knowledge of what a reasonable minimum size for their app is. Anything we picked would be essentially random.

**Comment by timsneath** [comment]
Oh, I think I was able to resize _beyond_ 0 pixels wide. I saw this on stock Ubuntu; let me see if I can capture a recording.

**Comment by stuartmorgan** [comment]
Does this still happen with the GTK embedding?

**Comment by timsneath** [comment]
Just tried it on the latest master with GTK embedder, but there's a different problem that is making it hard to reproduce -- specifically the `kFlutterWindowWidth` and `kFlutterWindowHeight` variables in `windows_configuration.cc` seem to be setting the minimum width and height (i.e. 1280x720 is the smallest it will go). That's not right, is it? Those should be setting the initial height and width only, surely?

**Comment by stuartmorgan** [comment]
Right, sorry; you'll need to re-test after https://github.com/flutter/flutter/pull/59630 lands (or just apply that change to your app directly).

**Comment by stuartmorgan** [comment]
Any update on reproducibility now that the can't-shrink-below-default-size fix is in?

**Comment by timsneath** [comment]
Tried to reproduce this morning but ran into #62146. Waiting on a resolution and then will try again.

**Comment by gspencergoog** [comment]
How about now @timsneath?  Hopefully the cosmic rays will be aligned well enough.

**Comment by timsneath** [comment]
You'll be pleased to know that indeed, the cosmic rays have finally aligned 👍 

And even more delightfully, this seems to be fixed in 1.21.0-10.0.pre.67. 

Thank you!
